packages/framework/src/Component/UploadedFile/UploadedFileRepository.php

```diff
@@ -81,4 +81,28 @@ public function getById(int $uploadedFileId): UploadedFile
 
         return $uploadedFile;
     }
+
+    /**
+     * @param int $uploadedFileId
+     * @param string $uploadedFileSlug
+     * @param string $uploadedFileExtension
+     * @return \Shopsys\FrameworkBundle\Component\UploadedFile\UploadedFile
+     */
+    public function getByIdSlugAndExtension(int $uploadedFileId, string $uploadedFileSlug, string $uploadedFileExtension): UploadedFile
+    {
+        $uploadedFile = $this->getUploadedFileRepository()->findOneBy(
+            [
+                'id' => $uploadedFileId,
+                'slug' => $uploadedFileSlug,
+                'extension' => $uploadedFileExtension,
+            ]
+        );
+
+        if ($uploadedFile === null) {
+            $message = 'UploadedFile with ID "' . $uploadedFileId . '", slug "' . $uploadedFileSlug . '" and extension "' . $uploadedFileExtension . '" does not exist.';
```

Comment:
Nitpick: I would preffer using `sprintf()` here. The absolutely best for me is putting those directly into exception's argument.

Reply:
Done